### PR TITLE
fix (cli): kaggle benchmark tasks

### LINKED ISSUE #984: Episode download fails for chunked serverside response
### Description
The Kaggle CLI crashes with a `KeyError: 'content-length'` when attempting to download competition episode replays. This is a regression of a known logic flaw previously identified in the related `Kaggle/kaggle-cli` repository (Issue [#357](https://github.com/Kaggle/kaggle-cli/issues/357)).

The crash occurs because the library assumes the `Content-Length` header is mandatory for all downloads. However, the server responds with `Transfer-Encoding: chunked` for dynamic content like episode replays, which, in line with HTTP/1.1 specifications, omits the `Content-Length` header.

### Environment
- **Python Version:** 3.13
- **Library:** `kaggle` (2.1.0)
- **OS:** macOS / Linux
- **Invokation** `kaggle competitions replay 75451155`

### Traceback
```text
  File "/.../site-packages/kaggle/api/kaggle_api_extended.py", line 2994, in download_file
    size = int(response.headers["Content-Length"])
  File "/.../site-packages/requests/structures.py", line 52, in __getitem__
    return self._store[key.lower()][1]
KeyError: 'content-length'
```

### Evidence (Header Dump)
Manual inspection of the response headers via `response.headers._store.keys()` confirms that `content-length` is absent and `transfer-encoding` is present:

odict_keys(['content-type', 'date', 'access-control-allow-credentials', 'access-control-allow-origin', 'content-encoding', 'set-cookie', 'vary', 'x-kaggle-millisecondselapsed', 'x-kaggle-requestid', 'x-kaggle-apiversion', 'x-kaggle-hubversion', 'x-frame-options', 'content-disposition', 'strict-transport-security', 'content-security-policy', 'x-content-type-options', 'referrer-policy', 'via', 'alt-svc', 'transfer-encoding'])

### Root Cause
In `kaggle/api/kaggle_api_extended.py`, the code performs a direct dictionary lookup on the header:
`size = int(response.headers["Content-Length"])`

Since the response is streamed (chunked), the key does not exist, leading to an immediate crash before the file can be saved.

### Suggested Fix
The `requests` library itself uses a [`prepare_content_length`](https://docs.python-requests.org/en/latest/api/#requests.PreparedRequest.prepare_content_length) approach to ensure headers are set to `"0"` for empty bodies or handled appropriately when lengths are unknown. The Kaggle CLI could implement a similar helper method to validate response headers before attempting to access them, for example (in `kaggle/api/kaggle_api_extended.py`):

```python
# Proposed helper method in KaggleApi class
def _prepare_content_length(self, response):
    content_length = response.headers.get("Content-Length")
    transfer_encoding = response.headers.get("Transfer-Encoding")

    if content_length is not None:
        return int(content_length)
    
    # Per RFC 7230, chunked encoding implies unknown content length.
    # We return 0 here to allow the stream to proceed.
    if transfer_encoding == 'chunked':
        return 0
        
    # Fallback: if neither header is present, we return 0 to allow the 
    # download to proceed via connection closure, though this signals 
    # a non-standard server response.
    return 0

# Updated line 2994 in kaggle/api/kaggle_api_extended.py:
size = self._prepare_content_length(response)
```

This approach follows the logic  in`requests`, where the absence of a length header is handled as a specific state rather than a failure point.

### PR DESCRIPTION
**Pull request by dolaameng**
Fixes and improves the benchmarks CLI:

1. **Fix `download_file` crash on chunked responses** (fixes #984): `Content-Length` is now read via `.get()`; size verification and resume logic gracefully handle `None`.

2. **Overhaul `download` command**: Outputs are auto-extracted from zip into a hierarchical `{task}/{model}/{run_id}/` layout. The command now validates the task exists, skips already-downloaded outputs, handles corrupt zips, and prints helpful hints when no runs are available.

3. **Robust task name parsing**: `_get_task_names_from_file` now supports positional `@task("name")` syntax and strips IPython/Jupyter magics (`%`, `!`, `%%`) before AST parsing so Jupytext percent-format files work correctly.

4. **Model slug `@`-fallback**: Client-side model filtering also matches `anthropic/claude-sonnet-4-6@default` → `claude-sonnet-4-6-default` as a workaround until the server returns the correct slug format.

5. **Better `push` behavior**: Pending tasks are now waited on (with `--wait`) instead of always erroring; a status-check hint is printed when `--wait` is omitted. Slug normalization warnings are shown when the task name differs from its slug.

6. **Improved output formatting**: Run errors are displayed in a dedicated section below the status table instead of inline. Enum values and task names are cleaned up in error messages.

7. **CLI polish & refactors**: `push`/`run` use proper "required arguments" groups with explicit usage strings. Pagination, URL construction, and notebook conversion are extracted into reusable helpers.

8. **More tests**: ~new/updated tests covering task name detection, IPython magic stripping, download edge cases, model slug fallback, and the `models` CLI.

### CODE REVIEW

File: src/kaggle/api/kaggle_api_extended.py [hunk 1]
```diff
@@ -3169,7 +3169,8 @@ def download_file(
             os.makedirs(outpath)
 
         # Get file metadata
-        size = int(response.headers["Content-Length"])
+        content_length = response.headers.get("Content-Length")
+        size = int(content_length) if content_length is not None else None
```
Comment (stevemessick) [inline src/kaggle/api/kaggle_api_extended.py]: Could be simply "if content_length", here and elsewhere.

Comment (dolaameng) [reply src/kaggle/api/kaggle_api_extended.py]: Thanks! It makes sense and also guides against empty "content_length".

By elsewhere do you mean like `if size is not None`? Would probably keep it since size now an int, `is not None` is more explicit?

**Review by stevemessick — CHANGES_REQUESTED** [review]
I don't see https://github.com/Kaggle/kaggle-cli/tree/main/docs/benchmarks.md. Gemini and Claude both need good docs. Also considering adding tutorials to tutorials.md. The ones I added were written by Gemini.

**Review by stevemessick — APPROVED** [review]
Almost there! The new docs file needs to be linked from README.md and index.rst. I'll go ahead and approve anyway. Thanks!

**Comment by dolaameng** [comment]
> Almost there! The new docs file needs to be linked from README.md and index.rst. I'll go ahead and approve anyway. Thanks!

Done. Thanks for the review and patience!
